RIOT-OS/RIOT
# drivers: modbus_rtu: add initial support

### Contribution description
This PR is a reboot of #13913 that got stale. I cloned it, rebased it onto master, adapted it to latest standards (hopefully) and extended it.

Contrary to the original PR, I removed a lot of `#ifdefs` that limited which Modbus functions would be supported. It was hard to manage IMHO.

I also extended the test application to test using a single or multiple devices.

### Testing procedure
A test application is provided in `tests/drivers/modbus_rtu`.

There are multiple ways to test this. The easiest is to connect RX/TX of one UART to TX/RX of another. That way, Modbus runs over 'RS-232'. Alternatively, connect two or more boards via a RS-485 network. Let one run as master and the other as slaves.

See [here](https://github.com/basilfx/RIOT/tree/feature/modbus_rtu/tests/drivers/modbus_rtu#modbus-rtu-driver-test) for more information.

<img src="https://github.com/RIOT-OS/RIOT/assets/815976/a5ec5081-47a5-490b-acb2-f8bc4e1c17b5" width="256">

### Issues/PRs references
Fixes #13869.
